types/types.go

```diff
@@ -164,6 +164,8 @@ type SpecReport struct {
 	CapturedStdOutErr string
 
 	// ReportEntries contains any reports added via `AddReportEntry`
+	// `AddReportEntry` supports adding entries concurrently, but direct read
+	// access to the slice is not thread-safe.
 	ReportEntries ReportEntries
```

Comment:
Would have liked to synchronize both read and write access at this level, but couldn't think of how to do it without introducing breaking changes.